server/src/frontend_api.rs

```diff
@@ -746,20 +747,16 @@ pub fn get_all_features(
     edge_token: EdgeToken,
     engine_cache: Data<DashMap<String, EngineState>>,
     token_cache: Data<DashMap<String, EdgeToken>>,
-    query_string: &str,
+    context: &Context,
     client_ip: Option<&ClientIp>,
 ) -> EdgeJsonResult<FrontendResult> {
-    let raw_context: IncomingContext = serde_qs::Config::new(0, false)
-        .deserialize_str(query_string)
-        .map_err(|_| ContextParseError)?;
-    let context: Context = raw_context.into();
     let context_with_ip = if context.remote_address.is_none() {
         Context {
             remote_address: client_ip.map(|ip| ip.to_string()),
-            ..context
+            ..context.clone()
         }
     } else {
-        context
+        context.clone()
```

Comment:
I wonder if we do have to clone here, there's no risk other than a bit more of memory allocation

Reply:
The clone here is pretty light, in this context it's probably over optimizing to fix given the amount of memory that's consumed in processing an HTTP request

...but it looks dumb and it's easy to fix so yeah, I took it out! Good eyes!